src/Installer/pkg/dotnet-sdk.csproj

```diff
@@ -0,0 +1,127 @@
+      SourceFiles="@(TemplatesFiles)"
+      OverwriteReadOnlyFiles="True"
+      SkipUnchangedFiles="False"
+      UseHardlinksIfPossible="False" />
+
+    <!-- Create layout: Workload Manifests -->
+    <Copy
+      DestinationFiles="@(ManifestFiles->'$(OutputPath)/sdk-manifests/%(RecursiveDir)%(Filename)%(Extension)')"
+      SourceFiles="@(ManifestFiles)"
+      OverwriteReadOnlyFiles="True"
+      SkipUnchangedFiles="False"
+      UseHardlinksIfPossible="False" />
+  </Target>
+
+  <Target Name="SetCustomPackagingProperties" BeforeTargets="_GetCommonJsonProperties">
+    <PropertyGroup>
+      <PackageSummary>Microsoft .NET SDK $(PackageVersion)</PackageSummary>
+      <!-- Clear PackageDescription to have it correctly set by Arcade. -->
+      <PackageDescription></PackageDescription>
+    </PropertyGroup>
+
+    <!--
+      Manpages are not installed atm, tracked with https://github.com/dotnet/sdk/issues/44833
+      Arcade infra expects the following items, but the installation path is TBD.
+      Tracked in https://github.com/dotnet/arcade/issues/15243
+
+    <ItemGroup>
+      <Manpage Include="$(ManpagesDirectory)**/*" />
+    </ItemGroup>
+    -->
+
+  </Target>
+
+  <Target Name="AddLinuxPackageInformation"
+      BeforeTargets="GetDebInstallerJsonProperties;GetRpmInstallerJsonProperties"
+      DependsOnTargets="CalculateLinuxNativeInstallerDependencyVersions">
+
+    <PropertyGroup>
+      <DebianPostinstTemplateFile>$(MSBuildThisFileDirectory)../redist-installer/packaging/deb/postinst</DebianPostinstTemplateFile>
+      <DebianPostinstFile>$(BaseIntermediateOutputPath)debian/postinst</DebianPostinstFile>
+    </PropertyGroup>
+
+    <!-- Generate postinst file -->
+    <ReplaceFileContents
+      InputFiles="$(DebianPostinstTemplateFile)"
+      DestinationFiles="$(DebianPostinstFile)"
+      ReplacementPatterns="%SDK_VERSION%"
+      ReplacementStrings="$(Version)" />
+
+    <ItemGroup>
+      <LinuxPackageDependency Include="dotnet-runtime-$(MicrosoftNETCoreAppMajorMinorVersion)" Version="$(MicrosoftNETCoreAppRuntimePackageVersionWithTilde)" />
+      <LinuxPackageDependency Include="dotnet-targeting-pack-$(MicrosoftNETCoreAppMajorMinorVersion)" Version="$(MicrosoftNETCoreAppRefPackageVersionWithTilde)" />
+      <LinuxPackageDependency Include="dotnet-apphost-pack-$(MicrosoftNETCoreAppMajorMinorVersion)" Version="$(MicrosoftNETCoreAppRuntimePackageVersionWithTilde)" />
+      <LinuxPackageDependency Include="netstandard-targeting-pack-$(NetStandardTargetingPackMajorMinorVersion)" Version="$(NetStandardTargetingPackPackageVersionWithTilde)" />
+      <LinuxPackageDependency Include="aspnetcore-runtime-$(AspNetCoreMajorMinorVersion)" Version="$(AspNetCoreRuntimeVersionWithTilde)" />
+      <LinuxPackageDependency Include="aspnetcore-targeting-pack-$(AspNetCoreMajorMinorVersion)" Version="$(AspNetCoreRefVersionWithTilde)" />
+      <LinuxPostInstallScript Include="$(DebianPostinstFile)" />
+    </ItemGroup>
+  </Target>
```

Comment:
Can we move most of this target out of a target and move the "replace file contents" task usage into a target that runs before the `GenerateInstallers` target when `BuildDebPackage` is true?

Reply:
Some of these properties are similarly set in `runtime`, in the same target, i.e. https://github.com/dotnet/runtime/blob/main/src/installer/pkg/sfx/installers/dotnet-host.proj

I agree that task should be conditioned to run for DEB build only.

I'll check what else can be moved out of this target.

Reply:
Yes, many of them are due to how the old tooling worked and how it was easier to reuse the existing spot.

Other ones are mainly around getting the current installer package version and using that to declare dependencies on other packages.

I'm working on some changes in arcade to make moving the second group out of a target possible.